### DISCUSSION TITLE
Pagination total number of pages

### DISCUSSION THREAD

**Discussion by erasebegin** [prisma/prisma#16478] [Q&A] [answered]
I feel this may be a very silly question, but I'm quite confused by this:

I want to use Prisma pagination so that less data is fetched when a page loads and it's great for that, but in order to show the user how many pages there are, I would need to know how many rows there are in total. But if I'm fetching all rows just to count them, I might as well just be delivering all rows to the frontend on each page load anyway.... am I missing something here? 

I guess my question is: **How do I get the total number of rows in a table without fetching all of the data in that table?**
I see that Prisma has a `select` feature so that I could fetch just one column such as ids of all rows in the table and then count those... presumbaly that would be more performant? Is that the way it is normally done?

Thank you for reading 🙏

**Comment by mithatercan** [comment, marked as answer]
Hello @erasebegin ,

Did you try to use [count](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#examples-14) key?  And [this discussion](https://github.com/prisma/prisma/discussions/3087) may be useful to you.

**Reply by erasebegin** [reply]
ahh, brilliant, thank you! I did search the questions here but I didn't think to use "count" as a search term 🙏

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
